init/onboard-central-ledger/onboard-dfsp.js

```diff
@@ -0,0 +1,499 @@
+        log(`EXE: FAIL: sendRequest->addDfsp:\t\t\t${JSON.stringify(error)}`);
+        log('EXE: INFO: Allowed failure:\t\t\t\tProceeding with next request');
+      } else {
+        throw new Error(`Response not OK/2XX: ${JSON.stringify(error)}`);
+      }
+    }
+  } catch ({ message }) {
+    log(`EXE: FAIL: sendRequest->addDfsp:\t${message}`);
+    process.exitCode = 1;
+    return;
+  }
+
+  try {
+    log('EXE: INIT: sendRequest->addInitialPositionAndLimits');
+    const response = await sendRequest(addInitialPositionAndLimits({
+      dfspName,
+      dfspCurrency,
+      netDebitCap,
+      initialPosition,
+      authToken,
+      hostCentralLedger,
+      baseCentralLedgerAdmin,
+      fspiopSource,
+    }));
+    if (response.ok) {
+      log('EXE: SUCC: sendRequest->addInitialPositionAndLimits');
+    } else {
+      const error = await response.json();
+      // Allow re-registering
+      if (response.status === 500 && error.errorInformation.errorCode === '2001'
+      && /already/.test(error.errorInformation.errorDescription)) {
+        log(`EXE: FAIL: sendRequest->addInitialPositionAndLimits:\t${JSON.stringify(error)}`);
+        log('EXE: INFO: Allowed failure:\t\t\t\tProceeding with next request');
+      } else {
+        throw new Error(`Response not OK/2XX: ${JSON.stringify(error)}`);
+      }
+    }
+  } catch ({ message }) {
+    log(`EXE: FAIL: sendRequest->addInitialPositionAndLimits:\t${message}`);
+    process.exitCode = 1;
+    return;
+  }
+
+  try {
+    log('EXE: INIT: sendRequest->getDfspAccounts');
+    const response = await sendRequest(getDfspAccounts({
+      dfspName,
+      authToken,
+      hostCentralLedger,
+      baseCentralLedgerAdmin,
+      fspiopSource,
+    }));
+    if (response.ok) {
+      const dfspAccounts = await response.json();
+      log('EXE: SUCC: sendRequest->getDfspAccounts');
+      log(`LOC: dfspAccounts:\t\t\t\t\t${JSON.stringify(dfspAccounts)}`);
+      log('EXE: INIT: settlementAccountIdFromHubAccounts');
+      const settlementAccountId = settlementIdFromHubAccounts(dfspAccounts, dfspCurrency);
+      log('EXE: SUCC: settlementAccountIdFromHubAccounts');
+      log(`LOC: settlementAccountId:\t\t\t\t${settlementAccountId}`);
+      log('EXE: INIT: sendRequest->depositFunds');
+      const innerResponse = await sendRequest(depositFunds({
+        dfspName,
+        dfspCurrency,
+        amount,
+        transferId: uuid.v4(),
+        settlementAccountId,
+        authToken,
+        hostCentralLedger,
+        baseCentralLedgerAdmin,
+        fspiopSource,
+      }));
+      if (innerResponse.ok) {
+        log('EXE: SUCC: sendRequest->depositFunds');
+      } else {
+        const innerError = await innerResponse.json();
+        throw new Error(`Response not OK/2XX: ${JSON.stringify(innerError)}`);
+      }
+    } else {
+      const error = await response.json();
+      throw new Error(`Response not OK/2XX: ${JSON.stringify(error)}`);
+    }
```

Comment:
Personally, I prefer the
```
if (somethingFailed) {
  // bail;
}

// do the rest of the stuff
```
paradigm.
So in this case
```
if (!response.ok) {
  const blah = blahblah.json();
  throw error
}

// rest of stuff
```
I think the cognitive load is lower. But I would quite happily agree to disagree- especially considering this is likely to be fairly well tested. So if you have even a little inkling of "nah, can't be bothered", just don't reply to this and mark it done or whatever.

Reply:
I'd have to think about it a bit longer. I don't disagree that this probably started to allow unnecessary complexity in, but the biggest point against it is that whatever I wrote would have to undergo more testing to make sure I didn't regress functionality, like you said.